File: Chapter-2/Example_2-2.cpp
#include <opencv2/highgui/highgui.hpp>

// bad way
using namespace cv;

int main(int argc, char** argv) {
	Mat img = imread(argv[1],-1);
	
	if (img.empty()) return -1;

	namedWindow("Example2", cv::WINDOW_AUTOSIZE);
	imshow("Example2",img);
	waitKey(0);

	destroyWindow("Example2");

	return 0;
}

File: Chapter-2/Example_2-2.py
import cv2 as cv
import argparse

parser = argparse.ArgumentParser()
parser.add_argument("-i","--image",required=True,\
        help="Path to image")
args = vars(parser.parse_args())

img = cv.imread(args["image"])
if img is None:
    print("Image not found")
else:
    cv.namedWindow("Example2",cv.WINDOW_AUTOSIZE)
    cv.imshow("Example2",img)
    cv.waitKey(0)

    cv.destroyWindow("Example2")
